bulk-ingester/src/test/java/org/alfresco/hxi_connector/bulk_ingester/processor/BulkIngestionProcessorIntegrationTest.java

```diff
@@ -0,0 +1,203 @@
+                new Node("71b5b65b-d92a-4944-9403-48b7ebf8664c", new ContentInfo(1496650, "UTF-8", "application/pdf"), parseProperties("cm:name=porcupine.pdf", "cm:title=", "cm:autoVersion=true", "cm:versionType=MAJOR", "cm:versionLabel=1.0", "cm:autoVersionOnUpdateProps=false", "aspectsNames=[cm:generalclassifiable, cm:versionable, cm:author, cm:thumbnailModification, cm:titled, cm:taggable, rn:renditioned, cm:auditable]", "cm:categories=11dedf84-4ebb-431e-adbf-7e92b2792674", "type=cm:content", "cm:lastThumbnailModification=doclib:1708330181340", "cm:description=", "createdAt=1708330180", "cm:initialVersion=true", "createdByUserWithId=admin", "modifiedByUserWithId=admin")));
+
+        // when
+        bulkIngestionProcessor.process();
+
+        // then
+        testNodePublisher.assertPublishedNodes(categories);
+        testNodePublisher.assertPublishedNode(folder);
+        testNodePublisher.assertPublishedNode(textFile);
+        testNodePublisher.assertPublishedNodes(pdfFiles);
+    }
+
+    @TestConfiguration
+    public static class MockEventPublisherConfiguration
+    {
+        @Bean
+        @Primary
+        public TestNodePublisher nodePublisher()
+        {
+            return new TestNodePublisher();
+        }
+
+        @Bean
+        @Primary
+        public ApplicationManager applicationManager()
+        {
+            return mock();
+        }
+    }
+
+    @Slf4j
+    public static class TestNodePublisher implements NodePublisher
+    {
+        private final Map<String, Node> publishedNodes = new HashMap<>();
+
+        @Override
+        public void publish(Node node)
+        {
+            log.info("Publishing node {}", node.toString());
+
+            publishedNodes.put(node.nodeId(), node);
+        }
+
+        public void assertPublishedNodes(List<Node> nodes)
+        {
+            nodes.forEach(this::assertPublishedNode);
+        }
+
+        public void assertPublishedNode(Node node)
+        {
+            Node publishedNode = publishedNodes.get(node.nodeId());
+
+            assertNotNull(publishedNode, format("Node %s not published", node.nodeId()));
+
+            assertEquals(node.nodeId(), publishedNode.nodeId());
+            assertEquals(node.contentInfo(), publishedNode.contentInfo());
+            assertMapsEquals(node.properties(), publishedNode.properties());
```

Comment:
Maybe AssertJ's `assertThat(node.properties()).usingRecursiveComparison().ignoringCollectionOrder().isEqualTo(publishedNode.properties())` could be handy in this case 🤔

Reply:
The (first?) issue seems to be that we expect `cm:lastThumbnailModification` to be a Set, but it's actually a List (and the order of items in that list changes). This seems like a bug in the code, rather than something we should gloss over.

Reply:
I've created https://github.com/Alfresco/hxinsight-connector/pull/172 for this.